packages/tailwindcss/src/index.ts

```diff
@@ -605,8 +616,27 @@ async function parseCss(
     sources,
   })
 
-  for (let customVariant of customVariants) {
-    customVariant(designSystem)
+  for (let name of customVariants.keys()) {
+    // Pre-register the variant to ensure its position in the variant list is
+    // based on the order we see them in the CSS.
+    designSystem.variants.static(name, () => {})
```

Comment:
Open for suggestions here. Like maybe a `designSystem.variants.reserve(name)` but felt silly to introduce a new method just for this...

But my idea was to have the same behavior as-if you are overwriting internal variants that should maintain the sort order.

Reply:
There's an argument for keeping the position of the lastly defined `@custom-variant` because otherwise if you were relying on a library that now introduces a `@custom-variant` with the same name, the sort order will be different.